packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -584,10 +599,44 @@ export class CfnInclude extends core.CfnElement {
     return cfnCondition;
   }
 
+  private getPlaceholderID(): string {
+    let id = 'Placeholder';
+    for (let i = 0; i < 26; i++) {
+      id = id + randomAlphaNumeric();
+    }
+    return id;
+
+    function randomAlphaNumeric(): string {
+      if (Math.random() < 0.5) {
+        return String.fromCharCode('a'.charCodeAt(0) + Math.floor(Math.random() * 26));
+      } else {
+        return Math.floor(Math.random() * 10).toString();
+      }
+    }
+  }
+
   private getOrCreateResource(logicalId: string, cycleChain: string[] = []): core.CfnResource {
     cycleChain = cycleChain.concat([logicalId]);
     if (cycleChain.length !== new Set(cycleChain).size) {
-      throw new Error(`Found a cycle between resources in the template: ${cycleChain.join(' depends on ')}`);
+      if (!this.allowCyclicalReferences) {
+        throw new Error(`Found a cycle between resources in the template: ${cycleChain.join(' depends on ')}`);
+      }
+      let placeholderResourceAttributes: any = this.template.Resources[logicalId];
+      let placeholderId: string = this.getPlaceholderID();
+      //handle multiple cycles
+      if (this.logicalIdToPlaceholderMap.get(logicalId)) {
+        this.logicalIdToPlaceholderMap.get(logicalId)!.add(placeholderId);
+      } else {
+        this.logicalIdToPlaceholderMap.set(logicalId, new Set<string>([placeholderId]));
+      }
+      let placeholderInstance = new core.CfnResource(this, placeholderId, {
+        type: placeholderResourceAttributes.Type,
+        properties: {},
+      });
+      placeholderInstance.overrideLogicalId(placeholderId);
```

Comment:
I'm pretty sure you could immediately make this refer to the actual logical ID, no?

Reply:
No, when recursing back to the creation of the non-placeholder object results in a duplicate logicalId error being thrown.

Reply:
Reconfirmed for sanity's sake, it does cause duplicate logicalID issues.